### ISSUE TITLE
Add R_t for variable R values over time by county

### ISSUE THREAD

**Issue by ajkeith** [CHAD-Analytics/CHAD#10]
**Is your feature request related to a problem? Please describe.**

R<sub>t</sub> is the initial reproduction number R<sub>0</sub> over time. When this number falls below 1, it indicates the spread of the virus is decreasing. This number can be very beneficial as an indicator for different regions.

http://systrom.com/blog/the-metric-we-need-to-manage-covid-19/  

**Describe the solution you'd like**

A Solution and excellent visualization exists over at:
https://rt.live/

With the code and documentation at 
https://github.com/k-sys/covid-19/blob/master/Realtime%20R0.ipynb

Transitioning this methodology to the county level or selected region would be the goal for CHAD

**Alternative Solution**

The results by state are provided at 
https://d14wlfuexuxgcm.cloudfront.net/covid/rt.csv

This can be a backup plan with full citation given to the creator, if we can't manage to adapt for county level values

**Comment by ajkeith** [comment]
@mattburzenski : https://rt.live/
https://github.com/k-sys/covid-19/blob/master/Realtime%20R0.ipynb
The base code is in Python and the math looks pretty intense, but he's a breakdown of how Rt can be calculated and applied at the State level.

@SlicedBadge: I think I can follow this math and code (after brushing off some cobwebs). I am intrigued and may try to implement at the county level. One issue I foresee is that it uses historical cases per day. Do we want to just use confirmed cases or should we scale them by some factor because not all cases are being tested?

@galarcon0308: @SlicedBadge After reading through his code and process, I believe he is also using historical "confirmed" cases per day? Probably best to stick with the confirmed case number even though we both know that it is underrepresented. Best of luck!

@mattburzenski: Yesterday Patrick mentioned a package in R called "reticulate" that can source python code into R.

Also, Patrick defended yesterday and got his PhD!

@Auburngrads: The reticulate package is great, but it's comes with a substantial increase in the difficulty associated with managing the app infrastructure. Plus package management in Python is already sketchy to begin with.

Also, shinyapps.io has support for Python built in (up to 3.5).

**Comment by ajkeith** [comment]
@SlicedBadge I assigned to you since it sounds like you've started work on this but feel free to unassign (and others feel free to add yourselves and help out)

**Comment by Auburngrads** [comment]
I'm happy to work this.  Including Python would be easy enough through the `reticulate` package, but I'm concerned with getting all of the resources to the server where the app will be hosted.

Implementing at the county level we need to think this through - doing Bayesian updating for N counties may be computationally expensive.  There are a few options:

1. Use `reticulate` to write an R function that serves as a wrapper for the native Python code
2. Re-write the Python as R and use STAN or JAGS to speed things up
3. Re-write the Python code as C++ using Rcpp

Options 1) and 2) require that we ensure that the needed resources are properly uploaded and configured on the server

3) requires the most code re-writing but is more robust to the location of the server (and it's the one I find the most interesting)

**Comment by SlicedBadge** [comment]
@Auburngrads I began working this at https://github.com/SlicedBadge/covid-19 (haven't pushed anything up yet). I may be out of my league if I need to recode to speed things up, but for now I am just taking their python code and adding documentation/tailoring to county level calculations. Since this would only need to be calculated once a day, calculation speed may not be an issue.

**Comment by Auburngrads** [comment]
Okay, got it.

Are you thinking that the app would download the updated dataset each day?

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by SlicedBadge** [comment]
Yes, Rt would only need to be updated each day (or probably only every couple of days). The Rts for each county can then just be read in by CHAD through a flat file. Eventually if the code is fast enough, this could be done on the fly, but for IOC I don't think it is necessary.

**Comment by SlicedBadge** [comment]
I have been working on this off and on over at https://github.com/SlicedBadge/covid-19 (realtime_rt.py has all the code). One issue with this approach at the county level is that a lot of counties don't have many new cases each day. This results in a few derivative issues that I have had to work around:
1) The prep stage of the code includes cutting off data when the infection is below a certain case count per day. They used 25 for the state level, but I dropped that down to 5 new cases in a day for the county level.
2) Because of the cutoff (above), some counties never get an estimated R_t because they never reach the cutoff.
3) Even with the cutoff, some counties have experienced so few cases recently that the number of new cases per day has dropped to 0 (after rounding smoothed numbers). 0 new cases per day blows up the equations used to estimate R_t by causing division by 0. To avoid this issue, I am currently rounding all 0s up to 1.
